src/screens/DiscoveryScreenV2/fragment/TopicFragment.js

```diff
@@ -217,9 +217,9 @@ const TopicFragment = ({
     renderDiscoveryItem({
       from: isFirstTimeOpen
         ? FROM_FOLLOWED_TOPIC_INITIAL
-        : index > newMapFollowedTopics.length
-        ? FROM_UNFOLLOWED_TOPIC
-        : FROM_FOLLOWED_TOPIC,
+        : item.following || item.user_id_follower !== null
+        ? FROM_FOLLOWED_TOPIC
+        : FROM_UNFOLLOWED_TOPIC,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new logic for determining the `from` variable seems to be more accurate as it now checks if the user is following the topic or if the user is a follower. However, it's important to ensure that `item.following` and `item.user_id_follower` are always defined and have boolean values to avoid unexpected results. If these properties can be `undefined` or non-boolean, consider adding additional checks.
```diff
-         : item.following || item.user_id_follower !== null
+         : (item.following === true) || (item.user_id_follower !== null && item.user_id_follower !== undefined)
```
This will ensure that the condition only evaluates to true when `item.following` is explicitly `true` and `item.user_id_follower` is not `null` or `undefined`.


<!-- This is an auto-generated comment by OSS CodeRabbit -->